api.go

```diff
@@ -59,8 +59,30 @@ func NewAPI() *API {
 	}
 }
 
+// functionStates specifies the api functions that are valid for each
+// cluster state.
+var functionStates = map[string][]int{
```

Comment:
This is a terrible name. Plus, those are methods, not functions. I'll clean up the naming conventions here if the overall approach is acceptable.